File: C#/BackgroundApp/BackgroundCS/StartupTask.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Net.Http;
using System.Diagnostics;
using System.IO;
using System.Threading.Tasks;
using Windows.Storage;
using Windows.ApplicationModel.Background;

// The Background Application template is documented at http://go.microsoft.com/fwlink/?LinkID=533884&clcid=0x409

namespace BackgroundCS
{
    public sealed class StartupTask : IBackgroundTask
    {
        private BackgroundTaskDeferral _deferal;

        public void Run(IBackgroundTaskInstance taskInstance)
        {
            // If you start any asynchronous methods here, prevent the task
            // from closing prematurely by using BackgroundTaskDeferral as
            // described in http://aka.ms/backgroundtaskdeferral

            Debug.WriteLine("StartupTask.Run()");
            _deferal = taskInstance.GetDeferral();
            taskInstance.Canceled += TaskInstance_Canceled;
        }

        private void TaskInstance_Canceled(IBackgroundTaskInstance sender, BackgroundTaskCancellationReason reason)
        {
            Debug.WriteLine("StartupTask.TaskInstance_Canceled() - {0}", reason.ToString());
            _deferal.Complete();
        }
    }
}

File: VB.NET/BackgroundApp/BackgroundVB/StartupTask.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Net.Http
Imports Windows.ApplicationModel.Background

' The Background Application template is documented at http://go.microsoft.com/fwlink/?LinkID=533884&clcid=0x409

Public NotInheritable Class StartupTask
    Implements IBackgroundTask

    Public Sub Run(taskInstance As IBackgroundTaskInstance) Implements IBackgroundTask.Run

        ' If you start any asynchronous methods here, prevent the task
        ' from closing prematurely by using BackgroundTaskDeferral as
        ' described in http://aka.ms/backgroundtaskdeferral

        ' Simple variables
        Dim five As Integer = 5
        Dim ten As Integer = 10
        Dim zero As Integer = 0
        zero = 2 * five - ten

        ' Square roots
        Dim x As Integer = 4
        Dim total As Integer = 49
        total = 15 * System.Math.Sqrt(9) + x

        ' This is Windows 10 IoT
        Const targetplatform As String = "IoT"
        Dim hero As Integer
        hero = targetplatform

    End Sub
End Class
